docs/DeveloperGuide.md

```diff
@@ -2,21 +2,149 @@
+The below shows the high-level class diagram:
+
+![Expense and Income Structure](img_2.png)
+
+#### Command Parsing and Execution
+
+Each command extends from an abstract `ExpenseCommand` or `IncomeCommand` base class and overrides the `execute(...)` method.
+
+Example sequence diagram for the command `add Lunch 12.5 Food 2025-03-25`:
+
+![Add Expense Command Sequence](img_1.png)
+
+This design simplifies testing, separates concerns clearly, and is extensible to other commands (e.g., filtering by date range in the future).
+
+#### Key Implementation Choices
+
+- **Command Pattern:** Each command is encapsulated in its own class with a single `execute(...)` method. This makes it easier to test and extend in the future.
+- **Category Normalization:** Category strings are capitalized by the parser to ensure consistent matching.
+- **Robust Input Parsing:** The parser uses defensive programming to avoid crashing on invalid user input, providing feedback when inputs are incomplete or incorrect.
+
+#### Alternatives Considered
+
+- Instead of a separate command class for each operation, we considered a generic `Command` with a type string and arguments. However, this would reduce readability and make each command's logic more error-prone.
+- We considered storing all entries in a shared superclass like `FinancialEntry`, but this was not implemented as Expenses and Incomes may diverge in fields and behavior later.
+
+#### Future Enhancements
+
+- Support for recurring entries.
+- Advanced search and filtering (e.g., date ranges).
+- UI display of graphs or trends.
+
+
+### 2. Hongheng
+   Responsible for implementing the Budget and Saving management modules. This includes the Budget, Saving, BudgetList, SavingList, and the associated commands and parsers.
+
+The goal of these modules is to allow users to manage budgets and track savings efficiently, with support for adding, editing, deleting, listing, and viewing summary data.
+
+#### Budget and Saving Structure
+Each of the two entities — Budget and Saving — is represented by its own class and managed through its own container class (BudgetList, SavingList).
+
+The diagram below shows the high-level class structure:
+
+![BudgetManager.png](team/BudgetManager.png)
+
+
+Budget: Represents a budget allocation for a specific month, storing:
+
+month: YearMonth
+
+amount: double
+
+description: String (optional)
+
+Saving: Represents a saved fund entry, storing:
+
+date: LocalDate
+
+amount: double
+
+purpose: String
+
+BudgetList and SavingList: Handle collections of budgets and savings, providing utilities like filtering and summarizing.
+
+#### Command Parsing and Execution
+Each command related to Budget or Saving extends from a common Command base class and overrides the execute(Model model) method.
+
+#### Example sequence diagram for the command:
+add-budget m/2025-04 a/1000 d/April allowance
```

Comment:
codeblocks here would be great as well, so testers can directly copy your command over, eg
```
add-budget m/2025-04 a/1000 d/April allowance
```